src/main/webapp/profile/Profile.js

```diff
@@ -0,0 +1,90 @@
+import React, { Component } from 'react';
+import { p, div, span } from 'react-hyperscript-helpers';
+import { spinnerService } from "../util/spinner-service";
+import { User } from '../util/ajax';
+import { format } from 'date-fns';
+import './Profile.css';
+
+class Profile extends Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      user: {},
+    };
+  }
+
+  componentDidMount() {
+    this.init();
+  }
+
+  init = () => {
+    spinnerService.showAll();
+    User.getUserSession(component.getUserUrl).then(resp => {
+      let role = 'Standard Access';
+      if (resp.data.isAdmin || resp.data.isORSP || resp.data.isComplianceOffice) {
+        role = 'Admin';
+      } 
+      else if (resp.data.isViewer) {
+        role =  "Read Only Access";
+      }
```

Comment:
Instead of setting role as a state value, it would be more efficient to take this role logic and turn it into a function. Then you don't have to store it as a state value. Same thing with `lastLoginDate`. Its better to store as little as possible in state and rely on calculating values using functions that take a state value as input. In this case, all of this can be calculated from `state.user`.